src/gc/env/gcenv.os.h

```diff
@@ -47,6 +47,42 @@ struct GCThreadAffinity
     int Processor;
 };
 
+// An event is a synchronization object whose state can be set and reset
+// indicating that an event has occured. It is used pervasively throughout
+// the GC.
+class GCEvent {
+public:
+    // Closes the event. Attempting to use the event past calling CloseEvent
+    // is a logic error.
+    virtual void CloseEvent() = 0;
```

Comment:
This should not be virtual. The rest of the GCToOSInterface is not virtual either.